core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1407,4 +1407,13 @@ package object config {
     .bytesConf(ByteUnit.BYTE)
     .createOptional
 
+  private[spark] val TRANSFERTO_NUM_CALLS =
+    ConfigBuilder("spark.file.transferToZeroReturns")
+      .doc("To track the numbers of 0 return value of Java NIO's transferTo. " +
+        "In some occasional case, Spark's copyFileStreamNIO will always return zero and " +
+        "run into infinite loop, this configuration is used to track the number of return 0 and " +
+        "jump out of loop in time. This is an internal configuration, " +
+        "user shouldn't configure it normally")
+      .intConf
```

Comment:
@jerryshao, actually I think it might be better to consider more conservative way to note this is an internal configuration via `.internal()`.